### PR TITLE
Switching reading to DataSource V2

### PR DESCRIPTION
**Pull request by Blackmorse**

### CODE REVIEW

File: src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+class ClickhousePartitionReader(clickhouseReaderInfo: ClickhouseReaderInfo, clickhouseInputPartition: ClickhouseInputPartition) extends PartitionReader[InternalRow] with Logging {
+  private val fields = clickhouseReaderInfo.schema.fields.map(f => s"`${f.name}`").mkString(", ")
+  val sql = s"SELECT $fields FROM ${clickhouseReaderInfo.tableName}"
+
+  private val conn = new ClickHouseDriver().connect(clickhouseReaderInfo.url, new Properties())
+  private val stmt = conn.prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)
+  stmt.setFetchSize(300)
+  logInfo(s"statement fetch size set to: ${stmt.getFetchSize}")
```
Comment (WonderBeat) [inline src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala]: debug?

File: src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      clickhouseInputPartition = partition.asInstanceOf[ClickhouseInputPartition])
+}
+
+class ClickhousePartitionReader(clickhouseReaderInfo: ClickhouseReaderInfo, clickhouseInputPartition: ClickhouseInputPartition) extends PartitionReader[InternalRow] with Logging {
+  private val fields = clickhouseReaderInfo.schema.fields.map(f => s"`${f.name}`").mkString(", ")
+  val sql = s"SELECT $fields FROM ${clickhouseReaderInfo.tableName}"
+
+  private val conn = new ClickHouseDriver().connect(clickhouseReaderInfo.url, new Properties())
+  private val stmt = conn.prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)
+  stmt.setFetchSize(300)
```
Comment (WonderBeat) [inline src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala:25]: + `.setQueryTimeout(seconds);`

File: src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        stmt.close()
+      }
+    } catch {
+      case e: Exception => logWarning("Exception closing statement", e)
+    }
+    try {
+      if (null != conn) {
+        conn.close()
+      }
+      logInfo("closed connection")
```
Comment (WonderBeat) [inline src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala]: debug?

**Review by WonderBeat — APPROVED** [review]

File: src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala (hunk 2 above)
Comment (Blackmorse) [reply src/main/scala/com/blackmorse/spark/clickhouse/datasourcev2/ClickhousePartitionReader.scala:25]: Will do in the task dedicated to timeout's tuning

**Pull request merged**